rest/admin_api.go

```diff
@@ -317,10 +317,27 @@ func (h *handler) handleActiveTasks() error {
 func (h *handler) handleGetRawDoc() error {
 	h.assertAdminOnly()
 	docid := h.PathVar("docid")
+	includeDoc := h.getBoolQuery("include_doc")
+	redact := h.getOptBoolQuery("redact", true)
+	salt := h.getQuery("salt")
 	doc, err := h.db.GetDocument(docid, db.DocUnmarshalAll)
-	if doc != nil {
-		h.writeJSON(doc)
+
+	h.setHeader("Content-Type", "application/json")
+
+	if redact {
+		h.response.Write([]byte(`{"_sync":` + doc.SyncData.HashRedact(salt)))
+	} else {
+		h.response.Write([]byte(`{"_sync":`))
+		h.addJSON(doc.SyncData)
 	}
+
+	if includeDoc {
+		h.response.Write([]byte(`, "doc":`))
```

Comment:
Nesting the doc body under `"doc"` feels a little wrong to do here, given the raw document is typically at the root. Perhaps we could iterate over each element in doc.Body() and append it to the JSON instead?